backend/src/services/instance/instance.entity.ts

```diff
@@ -93,15 +101,17 @@ class Instance {
     createdBy,
     awsInstanceId,
     tenantId,
+    state,
   }: {
     name: string;
     keyPairId: string;
     username: string;
-    hostname: string;
+    hostname: null;
     operationSystemId: string;
     createdBy: string;
     awsInstanceId: string;
     tenantId: string;
+    state: InstanceState;
```

Comment:
Entity (not service!!) is a place where most business logic should be. You can find more info [in this beautiful article](https://khalilstemmler.com/articles/typescript-domain-driven-design/entities/). Entities can even be read by non-developers to understand business logic and problems, keep it in mind.
If you are passing some param to method in entity it means that this param is coming either from user (some frontend form for example) or from 3rd party integration (aws, ...). However, state is mostly internal information. You can check whether param is internal or external (editable by user) by finding/thinking about PUT edpoint that edits this param. As we don't have any dropdown list with instances states it means that this param is internal.